.github/workflows/build.yml

```diff
@@ -0,0 +1,246 @@
+name: build
+on: [push, pull_request]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  linux:
+    name: linux
+    runs-on: ubuntu-18.04
+    container: lmmsci/linux.gcc:18.04
+    env:
+      CMAKE_OPTS: -DUSE_WERROR=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_COMPILE_CACHE=ON
+      CCACHE_MAXSIZE: 500M
+      MAKEFLAGS: -j2
+    steps:
+    - name: Update and configure Git
+      run: |
+        add-apt-repository ppa:git-core/ppa
+        apt-get update
+        apt-get --yes install git
+        git config --global --add safe.directory /__w/lmms/lmms
+    - name: Check out
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        submodules: recursive
+    - name: Cache ccache data
+      uses: actions/cache@v3
+      with:
+        key: ccache-${{ github.job }}-${{ github.ref }}-${{ github.run_id }}
+        restore-keys: |
+          ccache-${{ github.job }}-${{ github.ref }}-
+          ccache-${{ github.job }}-
+        path: ~/.ccache
+    - name: Configure
+      run: |
+        source /opt/qt5*/bin/qt5*-env.sh || true
+        mkdir build && cd build
+        cmake .. $CMAKE_OPTS -DCMAKE_INSTALL_PREFIX=./install
+    - name: Build
+      run: cd build && make
+    - name: Build tests
+      run: cd build && make tests
+    - name: Run tests
+      run: build/tests/tests
+    - name: Package
+      run: |
+        cd build
+        make install
+        make appimage || (cat appimage.log && false)
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: linux
+        path: build/lmms-*.AppImage
+    - name: Print ccache statistics
+      run: |
+        echo "[ccache config]"
+        ccache -p
+        echo "[ccache stats]"
+        ccache -s
+  macos:
+    name: macos
+    runs-on: macos-11
+    env:
+      CMAKE_OPTS: -DUSE_WERROR=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_COMPILE_CACHE=ON
+      CCACHE_MAXSIZE: 500M
+      MAKEFLAGS: -j3
```

Comment:
It's funny that macos uses 3 jobs, and the other ones use 2. Was there a reason for the difference? Also, was there a reason to downgrade from 6 (in circleci) to 2/3?

Reply:
I'm matching the number of jobs to the number of CPUs specified at https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources.